Frameworks/AEPSwiftUI/Sources/ContentCards/ContentCardUI.swift

```diff
@@ -19,12 +20,50 @@ public class ContentCardUI: Identifiable {
     /// The underlying data model for the content card.
     let schemaData: ContentCardSchemaData
 
+    public let template: any ContentCardTemplate
+
     /// SwiftUI view that represents the content card
-    public lazy var view: some View = EmptyView()
+    public lazy var view: some View = {
+        // TODO: Make adjustments to remove AnyView
+        AnyView(template.view)
+    }()
+
+    /// Factory method to create a `ContentCardUI` instance based on the provided schema data.
+    /// - Parameter schemaData: The `ContentCardSchemaData` to be used for the content
+    /// - Returns: An initialized `ContentCardUI` instance, or `nil` if unable to create template from `schemaData`
+    static func createInstance(with schemaData: ContentCardSchemaData) -> ContentCardUI? {
+        // determine the appropriate template based on the template type
+        let template: (any ContentCardTemplate)? = {
+            switch schemaData.templateType {
+            case .smallImage:
+                return SmallImageTemplate(schemaData)
+            case .largeImage, .imageOnly, .unknown:
+                // Currently unsupported template types
+                return nil
+            }
+        }()
```

Comment:
throw this implementation into a static private method. it will be cleaner and easier for testing

Reply:
Yeap, I will keep them in a separately 👍🏼